upstairs/src/client.rs

```diff
@@ -968,9 +955,7 @@ impl DownstairsClient {
             DsState::New
             | DsState::WaitActive
             | DsState::WaitQuorum
-            | DsState::Reconcile
-            | DsState::Deactivated
```

Comment:
Agreed that this behavior changed slightly; I think we can ignore it here, because #1577 adds explicit logic to [bail out early](https://github.com/oxidecomputer/crucible/blob/43a1e3098befd1944f90426663dbe06bdf1e2a1c/upstairs/src/client.rs#L1273-L1283) if we receive a negotiation message while not in the `Connecting` state.